projects/rero/ng-core/src/lib/translate/translate-loader.spec.ts

```diff
@@ -17,56 +17,93 @@
+          }
+        })
+      ],
+      providers: [TranslateService]
+    });
+    translate = TestBed.get(TranslateService);
+    http = TestBed.get(HttpTestingController);
   });
 
-  it('should be created', () => {
-    const translateLoader = new TranslateLoader(new CoreConfigService());
-    expect(translateLoader).toBeTruthy();
+  afterEach(() => {
+    translate = undefined;
+    http = undefined;
+  });
 
+  it('should be able to provide TranslateLoader', () => {
+    expect(TranslateLoader).toBeDefined();
+    expect(translate.currentLoader).toBeDefined();
+    expect(translate.currentLoader instanceof TranslateLoader).toBeTruthy();
   });
 
-  it('should override translation', () => {
-    const config = {
-      languages: ['fr'],
-      customTranslations: {
-        fr: {
-          'your query': 'Traduction surchargée',
-        }
-      }
-    };
+  it('should be able to get translations', () => {
+    translate.use('en');
 
-    const translateLoader = new TranslateLoader(config as CoreConfigService);
-    translateLoader.getTranslation('fr').subscribe(translations => {
-      expect(translations['your query']).toBe(config.customTranslations.fr['your query']);
+    // mock response after the xhr request, otherwise it will be undefined
+    http.expectOne('/assets/i18n/en.json').flush({
+      TEST: 'This is a test',
+      TEST2: 'This is another test'
     });
-  });
 
-  it('should throw an error because translations for language not found', () => {
-    const config = {
-      languages: ['fr']
-    };
+    // this will request the translation from the backend because we use a static files loader for TranslateService
+    translate.get('TEST').subscribe((res: string) => {
+      expect(res).toEqual('This is a test');
+    });
 
-    const translateLoader = new TranslateLoader(config as CoreConfigService);
-    expect(() => translateLoader.getTranslation('pt')).toThrowError('Translations not found for lang "pt"');
+    // this will request the translation from downloaded translations without making a request to the backend
+    translate.get('TEST2').subscribe((res: string) => {
+      expect(res).toEqual('This is another test');
+    });
   });
 
-  it('should not find custom translation for "en" language', () => {
-    const config = {
-      languages: ['fr', 'en'],
-      customTranslations: {
-        fr: {
-          'Record deleted.': 'Enregistrement effacé',
-        }
-      }
-    };
+  it('should be able to get translations in french', () => {
+    translate.use('fr');
 
-    const translateLoader = new TranslateLoader(config as CoreConfigService);
-    translateLoader.getTranslation('en').subscribe(translations => {
-      expect(translations['Record deleted.']).toBe('Record deleted.');
+    // mock response after the xhr request, otherwise it will be undefined
+    http.expectOne('/assets/i18n/fr.json').flush({
+      search: 'Recherche avancée',
+      'does not exists': 'Existe pas'
```

Comment:
A detail but: "N'existe pas" ?

Reply:
I have a problem with simple quote. I will change 'does not exists' => 'not exists'